Userland/Libraries/LibRegex/RegexMatch.h

```diff
@@ -225,21 +225,34 @@ class RegexStringView {
 
     size_t length() const
     {
+        if (m_cached_length.has_value())
```

Comment:
huh, I'm surprised this was an improvement. The 2 views that are expensive to compute code point length for (UTF-8 and UTF-16) already cache it; and getting code unit length is a constant thing.

Reply:
One thing that *is* expensive today is that "length in code points" is not cached across regex executions. Many test262 tests run regexes on the same UTF-16 string repeatedly, but we make a new RegexStringView each time, which then has to re-measure the string.

Reply:
🤔 I recall it being a fairly small improvement, I wouldn't mind removing this and figuring out how to fix the thing AK mentioned

Reply:
@awesomekling: the length of the input string is actually not _that_ useful (it's only used if matching in /m mode), there are many intermediate substrings created, and they're really only used once over the lifetime of the regex object.

Reply:
I've removed the patch in question, it just avoided a function call so its effect was very minimal regardless.